_source/_posts/2019-12-12-tutorial-kotlin-beginners-guide.md

```diff
@@ -0,0 +1,491 @@
+> 
+> Copy the issuer (found under **API** > **Authorization Servers**), client ID, and client secret into `src/main/resources/application.properties`.
+> 
+```properties
+okta.oauth2.issuer=$issuer
+okta.oauth2.client-id=$clientId
+okta.oauth2.client-secret=$clientSecret
+```
+
+Edit `KotlinSpringBootApplication` located the package `com.okta.developer` to add a controller mapping that will print a welcome message in the browser window.
+
+```kotlin
+package com.okta.developer
+
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.runApplication
+import org.springframework.security.core.annotation.AuthenticationPrincipal
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RestController
+
+@SpringBootApplication
+@RestController
+class KotlinSpringBootApplication {
+
+    @GetMapping("/")
+    fun hello(@AuthenticationPrincipal authenticationToken: OAuth2AuthenticationToken): String {
+
+        return "Welcome ${authenticationToken.principal.attributes["name"]}"
+    }
+}
+
+fun main(args: Array<String>) {
+    runApplication<KotlinSpringBootApplication>(*args)
+}
+```
+
+As you can see, in the `hello` function, the authenticated user is passed in the `authenticationToken`. `@AuthenticationPrincipal` annotation from Spring Security helps to resolve the principal to the expected principal type. The welcome message is built using **String interpolation**, a Kotlin idiom for variable substitution inside strings.
+
+Run the application with the following command:
+```txt
+./mvnw spring-boot:run
+```
+Browse to`http://localhost:8080` and the application should start an OAuth 2.0  authentication code flow, redirecting to the Okta login page.
+
+{% img blog/tutorial-kotlin-beginners-guide/okta-sign-in-widget.png alt:"Okta Sign-In Widget screenshot" width:"400" %}{: .center-image }
+
+After the login, you should see the welcome message:
+
+```txt
+Welcome, Jimena Garbarino
+```
+
+You now have a secure application with just a few lines of Kotlin and the help of Spring Boot!
+
+## Learn More about Kotlin, Java, and Secure Authentication with these Tutorials
+
+I hope you could grasp how succinct and expressive Kotlin is and why developers are loving it. You can find all the tutorial code in repositories [kotlin-hello-world](https://github.com/oktadeveloper/kotlin-hello-world-example) and [kotlin-spring-boot](https://github.com/oktadeveloper/kotlin-spring-boot-tutorial) on Github. To learn more about Kotlin, check out the following links:
+
+- [Build a Basic CRUD App in Android with Kotlin
+](https://developer.okta.com/blog/2018/09/11/android-kotlin-crud)
+- [Build an Application with Spring Boot and Kotlin](https://developer.okta.com/blog/2019/09/17/build-a-spring-boot-kotlin-app)
+- [Kotlin Reference](https://kotlinlang.org/docs/reference/)
```

Comment:
Blog post links should be relative and the first one should be on one line, not two.